# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/glibc32/powerpc32le.patch
# Copyright (C) 2023 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

Do not use '-mmultiple' on little-endian PowerPC systems, since those
instructions do not work when the processor is in little-endian mode.
The exceptions are PPC740 and PPC750 which permit these instructions
in little-endian mode.

--- glibc-2.36/sysdeps/unix/sysv/linux/powerpc/powerpc32/clone.S.vanilla	2023-01-07 17:17:49.726720823 +0100
+++ glibc-2.36/sysdeps/unix/sysv/linux/powerpc/powerpc32/clone.S	2023-01-07 17:45:52.919816459 +0100
@@ -39,7 +39,10 @@
 	/* Set up stack frame for parent.  */
 	stwu	r1,-32(r1)
 	cfi_adjust_cfa_offset (32)
-	stmw	r28,16(r1)
+	stw	r28,16(r1)
+	stw	r29,20(r1)
+	stw	r30,24(r1)
+	stw	r31,28(r1)
 
 	/* Set up stack frame for child.  */
 	clrrwi	r4,r4,4
@@ -80,7 +83,11 @@
 
 L(parent):
 	/* Parent.  Restore registers & return.  */
-	lmw	r28,16(r1)
+	lwz	r28,16(r1)
+	lwz	r29,20(r1)
+	lwz	r30,24(r1)
+	lwz	r31,28(r1)
+
 	addi	r1,r1,32
 	bnslr+
 	b	__syscall_error@local
